tests/mltoolbox_magic/shell_process_tests.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations under
+# the License.
+
+"""Test the datalab prediction with MLToolbox models."""
+from __future__ import absolute_import
+from __future__ import print_function
+
+import logging
+import os
+import psutil
+import six
+import subprocess
+import sys
+import threading
+import time
+import unittest
+
+import google.datalab.contrib.mltoolbox._shell_process as _shell_process
+
+
+class TestShellProcess(unittest.TestCase):
+  """Tests for process managements used in MLToolbox magics."""
+
+  def setUp(self):
+    self._logger = logging.getLogger('TestStructuredDataLogger')
+    self._logger.setLevel(logging.DEBUG)
+    if not self._logger.handlers:
+      self._logger.addHandler(logging.StreamHandler(stream=sys.stdout))
+
+  def test_process(self):
+    """ Test starting a process and have it depend on another process.
+    """
+
+    process_to_wait_args = ['python', '-c', 'import time; time.sleep(10)']
```

Comment:
I don't think it matters. It should be killed before it is done, assuming the test gets to process_to_wait.kill(). Maybe add a comment saying this